File: backend/src/main/java/com/llm_service/llm_service/dto/Conversation.java
package com.llm_service.llm_service.dto;

import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.SuperBuilder;

@EqualsAndHashCode(callSuper = true)
@SuperBuilder(toBuilder = true)
@Data
public class Conversation extends Base {
    String title;
    List<Discussion> discussions;
}

File: frontend/src/models/Conversation.ts
import { Discussion } from "./Discussion.ts";
import { ConversationId, DiscussionId } from "./Id.ts";
import { Dates } from "./Dates.ts";

export interface Conversation extends Dates {
  id: ConversationId;
  title: string;
  discussions: Discussion[];
}

export interface ConversationCompact extends Dates {
  id: ConversationId;
  title: string;
  discussions: DiscussionId[];
}
